### ISSUE TITLE
Non-adjustable Timeout Preventing Large Download Preparations from Completing

### ISSUE THREAD

**Issue by SBattaglia-R7** [Velocidex/velociraptor#3004]
This issue was reported to me by a customer who was attempting to download KAPE collections.  They sent me screenshots of attempts to prep downloads for two flows that were 500-700 MB in size.  Both attempts timed out after 600 seconds. 

Using the timeout advice in https://docs.velociraptor.app/docs/server_automation/#running-server-queries, I tried a workaround with a new Artifact calling create_flow_download().  Then I created server collections targeting the large collections, with the timeout set to 1000 seconds.  The download prep still timed out after 600 seconds. (The presence of the "wait" parameter in the function was probably a hint that the download job was independent of the server flow job.  I tried with and without it set to true.) 

I don't know if it's common for collections of that size to hit the timeout, but for what it's worth I was able to reproduce the timeout  multiple times in a different org.  Is there an existing method to adjust the timeout for this function, or can we add one?

Artifact used for the test:

```
name: Server.Download.FlowResults
description: |
   This artifact prepares a download package for the specified clientid and flowid.
   It behaves the same as the download functions in the flow UI.

type: SERVER

parameters:
   - name: client_id
     description: Client ID to export.
   - name: flow_id
     description: The flow id to export.
   - name: password
     description: An optional password to encrypt the collection zip.
   - name: format
     description: Format to export (csv,json,csv_only) defaults to both (csv).
     default: csv 
   - name: expand_sparse
     description: If set we expand sparse files in the archive.	
     default: false
     
sources:
   - query: |
    
         LET pw <= if(condition=password, then=password, else='')
      
         SELECT create_flow_download(
            client_id=client_id,
            flow_id=flow_id,
            password=pw,
            format=format,
            expand_sparse=expand_sparse,
            wait=true
            )
         FROM scope()
```

**Comment by scudette** [comment]
It looks like we have a hard coded timeout set in the create flow download operation itself

https://github.com/Velocidex/velociraptor/blob/master/vql/server/downloads/downloads.go#L239

We should probably allow for that to be extended.

Although I can't think of why it would take that long to just compress the results. Does this server run on a very slow filesystem?

**Comment by SBattaglia-R7** [comment]
You noted this in another thread but for visibility -- yes, the storage in the customer org and my test org is slower in trade of scaling up.  In my own tests I get ~500 MB out of ~700-800 MB compressed before it times out.    Downloading smaller collections/files doesn't feel slow to me and doesn't time out.  If we can make the timeout adjustable in the function without any side effects (can it still be cancelled if it goes on too long?) I will revise the above Artifact to make it easier to discover to help with Kape/large collections and add it to the Exchange.

**Comment by scudette** [comment]
This same function is used in the GUI - this is actually the reason it has its own timeout and runs independently of the query because the GUI fires it off in the background and updates the view of the download creation. 

We probably do not want to add an arbitrarily long timeout here so probably the first step would be to move the timeout value into the config file so an admin can make an informed choice of raising it (but I think a default of 10 min is fair).

Mainly the issue is in speeding up the download creation to compensate for slow backend filesystems.

**Comment by scudette** [comment]
fixed by #3058
